### DISCUSSION TITLE
Does bleak support LE CODED PHY?

### DISCUSSION THREAD

**Discussion by rty813** [hbldh/bleak#877] [Q&A]

**Comment by dlech** [comment]
I think the remote device needs to request this. On Windows, there is possibly [an api](https://docs.microsoft.com/en-us/uwp/api/windows.devices.bluetooth.bluetoothledevice.requestpreferredconnectionparameters?view=winrt-22000#windows-devices-bluetooth-bluetoothledevice-requestpreferredconnectionparameters(windows-devices-bluetooth-bluetoothlepreferredconnectionparameters) for exerting some control over this, but I haven't seen APIs for other OSes yet.

It might be possible to add a boolean property to test "does the hardware support coded phy" though as there appears to be cross-platform APIs for this.

**Reply by rty813** [reply]
I'm not sure about the underlying implementation of bleak, but I can set the phy of Bluetooth through HCI command. Can't bleak use this?
![2022-07-12_09-38](https://user-images.githubusercontent.com/22488813/178389520-239a8bd3-a307-4470-a7f5-1df8c3782dd6.png)
![2022-07-12_09-38_1](https://user-images.githubusercontent.com/22488813/178389523-21b6af2b-ef07-4546-8a38-34a7e7c727c3.png)

**Reply by dlech** [reply]
No, Bleak cannot send HCI commands directly.
